stdlib/public/Darwin/Accelerate/vImage_Buffer.swift

```diff
@@ -0,0 +1,348 @@
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter error: Overwritten with the error code if the operation failed.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
+                 bitsPerPixel: UInt32,
+                 error: inout Int,
+                 flags options: vImage.Options = .noFlags) {
+        
+        self.init()
+        
+        if size.width < 0 || size.height < 0 {
+            error = kvImageInvalidParameter
+            return nil
+        }
+        let width = vImagePixelCount(size.width)
+        let height = vImagePixelCount(size.height)
+        
+        error = vImageBuffer_Init(&self,
+                                  height,
+                                  width,
+                                  bitsPerPixel,
+                                  options.flags)
+        
+        if error < kvImageNoError {
+            return nil
+        }
+    }
+    
+    /// Initializes a vImage buffer of a specified size.
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
+                 bitsPerPixel: UInt32,
+                 flags options: vImage.Options = .noFlags) {
+        
+        self.init()
+        
+        if size.width < 0 || size.height < 0 {
+            return nil
+        }
+        let width = vImagePixelCount(size.width)
+        let height = vImagePixelCount(size.height)
+        
+        let error = vImageBuffer_Init(&self,
+                                      height,
+                                      width,
+                                      bitsPerPixel,
+                                      options.flags)
+        
+        if error < kvImageNoError {
+            return nil
+        }
+    }
+    
+    public func free() {
+        Darwin.free(data)
+    }
+}
+
+// MARK: Core Graphics Support
+
+@available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+extension vImage_Buffer {
+    
+    /// Initialize a vImage buffer with the contents of a Core Graphics image, reporting any errors.
+    ///
+    /// - Parameter cgImage: A `CGImage` instance to be used as the source.
+    /// - Parameter error: Overwritten with the error code if the operation failed.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    ///
+    /// This function will instantiate and initialize a vImage buffer from a `CGImage` using a `CGImageFormat`
+    /// based on the provided image's properties.
+    public init?(cgImage: CGImage,
```

Comment:
`_ cgImage: CGImage`; this is effectively a conversion, and shouldn't have a parameter label. You don't need `error`, because the init is failable. More generally, adopt this pattern throughout--either fail (trap or return nil) *or* return an error code, but not both. I'm not even sure that `flags` pulls its weight here--the only flag that's semantically valid in this function is `kvImageDoNotTile`.

Reply:
WRT to parameter name, I was taking my lead from `UIImage` that has initializers from [CG](https://developer.apple.com/documentation/uikit/uiimage/1624090-init) and [CI](https://developer.apple.com/documentation/uikit/uiimage/1624114-init) images, both with named parameters. 

I originally only had failable inits,  but I had pushback from engineering who wanted an option to get the error code back to facilitate figuring out why a particular initialization had failed.

`vImageBuffer_InitWithCGImage` also supports `kvImagePrintDiagnosticsToConsole`, and my `flags` parameter comes with a default.

Reply:
If you want to produce a descriptive error, this should throw, rather than having multiple error options.

Reply:
Agreed. I've removed all instances of passing an inout error variable in favor of throwing a proper error.